### DISCUSSION TITLE
[XYChart] - Border radius on a barStack

### DISCUSSION THREAD

**Discussion by maximerivolet** [airbnb/visx#1093] [Q&A]
Hello,

I'm using XYChart to display barStacks. I would like to have a border radius :

- on the bottom of each first rect
- on the top of each last rect

My problem is that I can't manage to establish which rect are first and which rect are last on a barStack.

Do you have any advice for this ? 

Regards,

**Comment by williaster** [comment]
Hey @goldenivan 👋  currently this isn't possible, there's a related request in #1092 with a workaround to apply a border radius to **all** corners (which I realize is not what you want). I think we could add the functionality to all bar components (Bar, BarGroup, BarStack) since there seems to be the desire to modify the aesthetics in that way.

**Reply by williaster** [reply]
@goldenivan if we add this support, do you have any thoughts on the API for specifying which corners to round? seems like it'd be ideal if you specified them per `BarSeries` in a stack for maximal flexibility, e.g., 

```tsx
() => (
  <BarStack>
    <BarSeries {...} radius={4} topRadius />
    <BarSeries {...} radius={4} bottomRadius />
  </BarStack>
)
```

**Reply by williaster** [reply]
Added in #1098

**Reply by maximerivolet** [reply]
Hi @williaster, thank you for the answer. Setting the first and last element of a barStack to be rounded might be a good idea. It's an easy way to set this. I'll wait the merge request !

**Reply by maximerivolet** [reply]
Hi, after thinking about it, it might be complicated. For example, in our barStack, we have not every values on every bars. So the last bar is not always the same.

**Reply by williaster** [reply]
> in our barStack, we have not every values on every bars. So the last bar is not always the same.

ahhh, yes this is more complicated. Thanks for mentioning this! I think the probably means it should be set-able per bar through an accessor function or something. will hold off on merging #1098 so we don't have a subsequent breaking API change, will try to work in this requirement.

**Reply by maximerivolet** [reply]
Hi @williaster,

I saw that the PR #1098 was merged. I tested it, and this isn't applying the radius only for the bar on the top, but all bars. Is there a way to adjust that ?

**Reply by williaster** [reply]
hmm, you tried these props in order to get this effect?

<img src="https://user-images.githubusercontent.com/4496521/148832958-4f0c5365-0436-486a-a5c2-1531902fd3b2.png" width="500" />

**Reply by maximerivolet** [reply]
Hi @williaster ! In my case, I've got a BarStack with a single BarSeries, like this : 

```
<BarStack>
  {listSigles.map((sigle, i) => {
    return (
      <BarSeries
        key={`${sigle}-${i}`}
        dataKey={sigle}
        data={isRange && renderVertically ? dataAgeRange : dataAge}
        xAccessor={(data) =>
          renderVertically ? data.age : axisLeft ? data.groups[sigle].length : -data.groups[sigle].length
        }
        yAccessor={(data) => (renderVertically ? data.groups[sigle].length : data.age)}
        colorAccessor={() => getGroupColor(sigle, groups)}
      />
    )
  })}
</BarStack>
```

In my case, all bar series are not used on every bars. This is an example : 

![image](https://user-images.githubusercontent.com/12681339/150014219-e73ffa41-0cc0-458b-94f2-aa68fe35ecd2.png)

So, as I can't map all bar series, do you have a workaround for this case ?

**Reply by williaster** [reply]
hey @goldenivan , thanks for elaborating. I toyed around a bit with an example like that but unfortunately don't have a workaround as we didn't add per-bar support to the initial implementation. So this would require more work to support. Happy to review a PR if there is any interest.

**Comment by dvyio** [comment]
I happened to see this question after replying to [a similar issue](https://github.com/airbnb/visx/issues/1475) and thought it might help you too.

Can you use a rounded rectangle as a mask for each stack? (I go into slightly more detail in the issue reply.)

https://github.com/airbnb/visx/issues/1475

**Reply by maximerivolet** [reply]
The problem is doing that imply to know where is the rectangle at the top of each stacks. I think it would be very complicated.

**Reply by dvyio** [reply]
If I get time I'll see if I can code up an example that works. I haven't used `BarSeries` much so it'll be a good excuse to play with it.

**Reply by maximerivolet** [reply]
Be careful, **BarSeries** is a thing, but they are inside a **BarStack**, and we don't manage who is where, Visx is doing that for us.
